zcash_client_sqlite/src/lib.rs

```diff
@@ -1398,7 +1433,40 @@ impl<P: consensus::Parameters> WalletWrite for WalletDb<rusqlite::Connection, P>
                             None,
                         )?;
                     }
-                    _ => (),
+                    #[cfg(feature = "transparent-inputs")]
+                    Recipient::EphemeralTransparent {
+                        receiving_account,
+                        ephemeral_address,
+                        outpoint_metadata,
+                    } => {
+                        wallet::transparent::put_transparent_output(
+                            wdb.conn.0,
+                            &wdb.params,
+                            outpoint_metadata,
+                            &TxOut {
+                                value: output.value(),
+                                script_pubkey: ephemeral_address.script(),
+                            },
+                            None,
+                            ephemeral_address,
+                            *receiving_account,
+                        )?;
+                        wallet::transparent::mark_ephemeral_address_as_used(
+                            wdb,
+                            ephemeral_address,
+                            tx_ref,
+                        )?;
+                    }
+                    #[cfg(feature = "transparent-inputs")]
+                    Recipient::External(zcash_address, PoolType::Transparent) => {
+                        // Always reject sending to one of our ephemeral addresses.
+                        wallet::transparent::check_ephemeral_address(
```

Comment:
Should this be `check_is_not_ephemeral_address`? That's what the comment seems to imply, but it's not clear from the code what happens here.

Reply:
Yes the function is poorly named, partly because it does one of two things (check that an address is not ephemeral, or check whether an address is not a *used* ephemeral address) depending on whether the boolean argument is true. I can refactor it to avoid that.